spec/win32/unit/certstore/certstore_spec.rb

```diff
@@ -0,0 +1,124 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe Win32::Certstore do
+
+  let (:certstore) { Win32::Certstore }
+  let (:certbase) { Win32::Certstore::StoreBase }
+  
+  describe "#list" do
+    context "When passing empty certificate store name" do
+      let (:store_name) { "" }
+      it "Raise ArgumentError" do
+        expect { certstore.open(store_name) }.to raise_error(ArgumentError)
+      end
+    end
+
+    context "When passing invalid certificate store name" do
+      let (:store_name) { "Chef" }
+      it "Raise ArgumentError" do
+        expect { certstore.open(store_name) }.to raise_error(ArgumentError)
+      end
+    end
+
+    context "When passing empty certificate store name" do
+      let (:store_name) { nil }
+      it "Raise ArgumentError" do
+        expect { certstore.open(store_name) }.to raise_error(ArgumentError)
+      end
+    end
+
+    context "When passing valid certificate store name" do
+      let (:store_name) { "root" }
+      let (:root_certificate_name) { "Microsoft Root Certificate Authority"}
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:cert_list).and_return([root_certificate_name])
+      end
+      it "return certificate list" do
+        store = certstore.open(store_name)
+        certificate_list = store.list
+        expect(certificate_list.size).to eql(1)
+        expect(certificate_list.first).to eql root_certificate_name
+      end
+    end
+
+    context "When passing valid certificate store name" do
+      let (:store_name) { "root" }
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:cert_list).and_return([])
+      end
+      it "return no certificate list" do
+        store = certstore.open(store_name)
+        certificate_list = store.list
+        expect(certificate_list.size).to eql(0)
+      end
+    end
+
+    context "When adding invalid certificate" do
+      let (:store_name) { "root" }
+      let (:cert_file_path) { '.\win32\unit\assets\test.cer' }
+      it "return no certificate list" do
+        allow(certbase).to receive(:CertAddEncodedCertificateToStore).and_return(false)
+        store = certstore.open(store_name)
+        expect { store.add(cert_file_path) }.to raise_error(Chef::Exceptions::Win32APIError)
+      end
+    end
+
+    context "When adding certificate failed with FFI::LastError" do
+      let (:store_name) { "root" }
+      let (:cert_file_path) { '.\win32\unit\assets\test.cer' }
+      
+      it "return 'The operation was canceled by the user'" do
+        allow(certbase).to receive(:CertAddEncodedCertificateToStore).and_return(false)
+        allow(FFI::LastError).to receive(:error).and_return(1223)
+        store = certstore.open(store_name)
+        expect { store.add(cert_file_path) }.to raise_error(Chef::Exceptions::Win32APIError)
```

Comment:
It would be good to check this with the actual error message: 
```expect { store.add(cert_file_path) }.to raise_error('The operation was canceled by the user')```